src/asprof.h

```diff
@@ -37,6 +38,28 @@ typedef const char* (*asprof_error_str_t)(asprof_error_t err);
 DLLEXPORT asprof_error_t asprof_execute(const char* command, asprof_writer_t output_callback);
 typedef asprof_error_t (*asprof_execute_t)(const char* command, asprof_writer_t output_callback);
 
+// This API is UNSTABLE and might change or be removed in the next version of async-profiler.
+typedef struct {
+    // A thread-local sample counter, which increments (not necessarily by 1) every time a
+    // stack profiling sample is taken using a profiling signal.
+    //
+    // The counter might be initialized lazily, only starting counting from 0 the first time
+    // `asprof_unstable_get_thread_local_data` is called on a given thread. Further calls to
+    // `asprof_unstable_get_thread_local_data` on a given thread will of course not reset the counter.
+    volatile uint64_t sample_counter;
+} asprof_unstable_thread_local_data;
+
+// This API is UNSTABLE and might change or be removed in the next version of async-profiler.
+//
+// Gets a pointer to asprof's thread-local data structure, see `asprof_unstable_thread_local_data`'s
+// documentation for the details of each field. This function might lazily initialize that
+// structure.
+//
+// This function can return NULL in case of an allocation failure.
+//
+// This function is *not* async-signal safe.
+DLLEXPORT asprof_unstable_thread_local_data *asprof_unstable_get_thread_local_data(void);
```

Comment:
You've explicitly documented the function as unstable in the code and in the docs, which is totally fine; adding `unstable` in the name is going to be too much ) The name is already too long.